pallets/block-rewards/src/lib.rs

```diff
@@ -237,7 +239,10 @@ pub mod pallet {
 	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
 		fn build(&self) {
 			T::Rewards::attach_currency(
-				(T::Domain::get(), CfgCurrencyId::BlockRewards),
+				(
+					T::Domain::get(),
+					CfgCurrencyId::Staking(StakingCurrency::BlockRewards),
```

Comment:
Not sure if this has already been said. Why use this directly instead of a `CurrencyId` type parameter? I feel like the `StakingCurrency::BlockRewards` value should be added by config.

Reply:
Yes we can do this. I interpreted [your comment](https://github.com/centrifuge/centrifuge-chain/pull/1198#discussion_r1105795201) to include this for the `CurrencyId` as well.

Reply:
Thanks for the change @wischli! 🙌🏻